### ISSUE TITLE
Problem with static libs using ifort/icc on macOS

### ISSUE THREAD

**Issue by kgerheiser** [NOAA-EMC/NCEPLIBS#41]
I was having trouble building NCEPLIBS with ifort on my Mac as I was discussing with @mark-a-potts. 

The issue was linking the NCEP-post executable:

```
Undefined symbols for architecture x86_64:
_cmassi_mod_mp_cn0r0_ referenced from:
      _microinit_ in libnceppost.a(MICROINIT.F.o)
      _calmict_new_ in libnceppost.a(CALMICT.f.o)
      _calmict_old_ in libnceppost.a(CALMICT.f.o)
.....
```

So, it's missing the symbol `_cmassi_mod_mp_cn0r0` (among others), but then you look at the symbols in `libnceppost.a` by running nm:

```
$ nm libnceppost.a

libnceppost.a(CMASSI.f.o):
0000000000000040 s EH_frame0
0000000000000000 T _cmassi_mod._
0000000000000058 S _cmassi_mod._.eh
0000000000000004 C _cmassi_mod_mp_cn0r0_f_
...
```

and you clearly see the symbol right there in `CMASSI.f.o`, which is in the very same library.

So, why does the linker think the symbol in missing? It's just a regular (uninitialized) module variable located in `CMASSI.f

The key is the `C` in the nm output `0000000000000004 C _cmassi_mod_mp_cn0r0_f_`

That means the symbol is a `common symbol`, which basically means that it's an uninitialized value.

But `ar` is weird on macOS and doesn't include common symbols in the symbol table of the library, so when linking to `libnceppost` the linker can't see it.

The reason this all works with GCC is that in the symbol table instead of being a `C` the symbol is an `S` which is for `small data or uninitialized values` which the macOS archiver does see.

There are several ways to fix this: you can initialize the value to something, create the static library using `ifort -staticlib`, or pass the `-c` option to `ar` which unhides common symbols.

The third option is the best, and to do that in CMake you can modify how `ar` is called  and pass it `-c` by setting: 

`set(CMAKE_Fortran_ARCHIVE_FINISH "<CMAKE_RANLIB>  -c <TARGET>")`

As a side note this would also be a good place to add `-no_warning_for_no_symbols` to suppress the no symbol warnings you get on macOS.

I'll make a PR with this fix in both UFS Weather Model and this repo, but I think it was an interesting problem worth documenting.

**Comment by climbfuji** [comment]
Wow. Great investigation and explanation. Looking forward to your PRs. Thanks!

**Comment by kgerheiser** [comment]
Fixed via #43
